### PR TITLE
Move me to bottom nav

### PR DESCRIPTION
**Pull request by ravishanker**
Moves Me (avatar) to Bottom Navigation as a fourth tab after Notifications on JP app, and in menu under manage on WP app
- Updates MySite Info layout
- Updates Me screen layout

Fixes # pcdRpT-2HW-p2#comment-4875, and this discussion p1686812471360099-slack-C04SFL0RP51

**Jetpack app**
| Me at the Bottom | Me Screen | Dark Mode
|--------|--------|--------|
|Screenshot_20230618_160803|Screenshot_20230618_160742|Screenshot_20230618_162015|

**WordPress app**
| Me in menu | Me |
|--------|--------|
|Screenshot_20230620_171528 | Screenshot_20230620_171538|

To test:

- Launch Jetpack app, and login
- Verify that Me/Avatar doesn't appear at top right hand corner anymore 
- Verify that Me is in the bottom navigation, next to Notifications
- Tap on Me tab
- Verify that it goes to Me Screen as before

- Launch WordPress app, and login
- Verify that Me/Avatar doesn't appear at top right hand corner anymore 
- Verify that Me is in the menu, under manage category
- Tap on Me row
- Verify that it goes to Me Screen as before

#### To-do for later in a separate PR

- [ ] Replace icons with new ones on Me Screen
- [ ] Add change photo cta on Me -> My Profile screen if required

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Existing tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Review by osullivanchris — APPROVED** [review]
Sharing per slack the reasons I think this is a good move:

1. the iA is more 'correct'. Today, 'Me' is inside My Site, which is the opposite of the actual parent/child structure
2. its more consistent with WordPress.com where the top bar has My Sites, Reader, Me, Notifications
3. When we merged Me into My Site years ago, My Site was just a menu. Now My Site is more complicated and we have several rows of navigations/information at the top
4. We're using a lot of vertical space for only one icon 
5. Removing that vertical space aligns the header information on My Site, Reader, Notifications more consistently

Here are a couple of improvements I think we could make. Wouldn't have to do all immediately and some are separate issues. The padding bug is a must fix of course. Then, I think changing the layout of the profile photo and name on 'Me' to match the My Site header would be a big improvement, if I was to pick one change.

Me-Tab-Tweaks

**Comment by ravishanker** [comment]
👋 @ovitrif - I've added you as a reviewer since you're also doing some work in this area. 🙇

**Comment by ravishanker** [comment]
👋 @pachlava - I've added you as this might fail instrumented tests, any other automation stuff. 🙇

**Comment by AjeshRPai** [comment]
Hey @ravishanker 👋🏼 

Awesome job on moving me to the bottom nav 🙌🏼 . Sorry for jumping in, but I was doing some changes in the dashboard and noticed this PR. 

FYI - On **jetpack phase removal four**, **new users phase** and **self hosted users phase**, we are removing this bottom nav entirely to prevent the access of reader and notifications. you might need to make some changes to show the bottom nav without a reader and notifications but only with me page and my site bottom nav in the above phases. 

⚠️  🐛  I tested this PR and enabled the `jp_removal_four` flag, and the bottom nav is completely gone. So a user can't access the me page at all. Screenshot below. 



I also noticed the padding issue, but I am unsure if that's a WIP. If thats fixed, let me know so I can double-check. 

also, There is an issue int the enabling of phase 4 right now which we are trying to fix with https://github.com/wordpress-mobile/WordPress-Android/pull/18612, 


TO reproduce the issue of the me tab not shown, you can try any of the below scenarios
1. ~Disable `jp_removal_static_posters_phase` and enable `jp_removal_four`~
2. ~IF you are testing this scenario after this [PR](https://github.com/wordpress-mobile/wordpress-android/pull/18612) is merged, then just~ enable `jp_removal_four`
3. enable `jp_removal_new_users` - this phase is rolled out, in this phase notifications and reader is not shown 
4. enable `jp_removal_self_hosted_users`- this phase is rolled out as well, in this phase notifications and reader is not shown as well 

Let me know if you need any help with the changes that need to be done or want more information on this.

**Comment by osullivanchris** [comment]
@AjeshRPai thanks for taking the time to look and highlighting the changes on the WordPress side. That wasn't front in my mind. I'm going to take a look at it. But I think the 'default' solution would be having a tab bar with 2 tabs as you described.

Another approach would be to keep the old layout with 'me' in the My Site navigation bar only for WordPress, but that seems like more differences between the two apps to support and maintain than a tab bar with items which turn on and off...

**Comment by ravishanker** [comment]
> Let me know if you need any help with the changes that need to be done or want more information on this.

@AjeshRPai - Thanks for noticing, and flagging this early 🙇   I'll need to make some changes then in the light phase four.  I'll add you as a reviewer as well to make sure it all works before and after.  

This can be done in two ways
- Bottom nav with My Site, and Me
- Me under Manage in the menu (ignore profile pic, and padding for now)

| Bottom Nav | Manage |
|--------|--------|
| Screenshot_20230620_142832| Screenshot_20230620_141159|

**Comment by ovitrif** [comment]
> 👋 @ovitrif - I've added you as a reviewer since you're also doing some work in this area. 🙇

This is a no-brainer for me, I like it, though as @AjeshRPai shared, there is some overlap with ongoing work to improve the daily development workflow, by optimising and making debug settings easily accessible regardless of logged in/out state.

Then there's also the phase 4, which Chris is aware of.

Thus carry on from my side 🙏🏻 , I'm not needed here since I ended up focusing on crashes and didn't do any Compose refactoring 😅 .

File: WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java [hunk 1]
```diff
@@ -663,14 +660,6 @@ private void initViewModel() {
...
             String message = getResources().getString(fabUiState.getCreateContentMessageId());
 
-            mFabTooltip.setMessage(message);
-
-            if (fabUiState.isFabTooltipVisible()) {
-                mFabTooltip.show();
-            } else {
-                mFabTooltip.hide();
-            }
-
```
Comment (AjeshRPai) [inline WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java:673]: 👏🏼   thanks for removing this tooltip, ICYMI you can close [this issue](https://github.com/wordpress-mobile/wordpress-android/issues/18249) once this PR is merged 
cc: @thomashorta

**Review by AjeshRPai — CHANGES_REQUESTED** [review]
Hey @ravishanker 

I tested the following scenarios 
- migration from static posters phase to phase 4| self hosted users phase | new users phase 🟢 
    -  Wordpress app 
       -  Bottom nav is not shown on phase 4 and me item is shown in menu list 🟢 
       - if the user is on me fragment on static posters phase and the app goes to background, when the phase 4 is switched on user is navigated to my site dashboard without bottom nav. 🟢 
   -  Jetpack app 
       - no ui changes when user migrates 🟢 
- if me tab was the last opened tab, when a user opens the app again, me tab is shown 🟢 

Minor issues 
- In wordpress app (static posters phase) me option is show on the menu items as well on bottom nav - as per my understanding it should be shown either on menu items or on bottom nav, correct me if am wrong 
- in both wp/jp app people icon is a dark circle/white circle 

|  light mode | dark mode | 
|---|---|
| screenshot_my_site_wordpress__  | screenshot_my_site_wordpress_darkmode__  | 

Other than these minor issues, all looks good to me 👍🏼 

cc: @zwarm 
1. for letting you about this change on dashboard, in case if you want to test for any regression issues 
2. also for letting you know that I have tested to make sure that there are no regression issues that we fixed for phase 4.  mainly 👇🏼 
   1. https://github.com/wordpress-mobile/WordPress-Android/issues/17811 
   2. https://github.com/wordpress-mobile/WordPress-Android/pull/17828

**Comment by tiagomar** [comment]
Hey @ravishanker, it seems like we could save some vertical space in Landscape mode.

image

**Comment by tiagomar** [comment]
🐛  When selecting `Me` tab, TalkBack reads the `My Site` description.

https://github.com/wordpress-mobile/WordPress-Android/assets/42008628/49453555-cc28-4638-91ad-fee698840832

**Comment by ravishanker** [comment]
> Minor issues
> 
> * In wordpress app (static posters phase) me option is show on the menu items as well on bottom nav - as per my understanding it should be shown either on menu items or on bottom nav, correct me if am wrong
> * in both wp/jp app people icon is a dark circle/white circle
> 
👋 @AjeshRPai - Thanks for testing for various phases.  Fixed the those minor issues.  Please review again. 🙇

**Comment by ravishanker** [comment]
> Hey @ravishanker, it seems like we could save some vertical space in Landscape mode.

👋 @tiagomar - This is happening on JP app only in landscape mode right?  It is proving difficult fix this due to various feature flags and differences between WP/JP etc.  Need to handle separately perhaps afterwards

**Comment by ravishanker** [comment]
> 🐛 When selecting `Me` tab, TalkBack reads the `My Site` description.
> 
>  meTab.mp4

Me tab is reading the content description correctly.  When on Me screen, it is reading the MySite description.  Still investigating where that is coming from!

**Comment by ravishanker** [comment]
👋 @pachlava - If you could please take a look at the failing instrumentation tests, if you're back from AFK . 🙇

**Comment by pachlava** [comment]
I'm taking a look @ravishanker 👋

**Comment by pachlava** [comment]
@ravishanker 👋 

A 0db191214b702bcccc7a4ddfbf001ba7ed303b71 of yours updates the flow to get _to_ the `Me` screen 👍, however, there remained one thing that also needed an update: a way to go _back_ from `Me` screen.

Previously, this was done by pressing the back button, since the navbar was not available for `Me` screen, and it resulted in `My site` screen being shown. Now, however, pressing back results in the app being minimized. So the back button press had to be replaced with tapping `My Sites`, which I did in 744b3fe874b6b81a5974e72a3fa7b2a5d79ed657. All the tests passed locally for me, 🤞 for the 🟢 Buildkite.

Apparently, the only place where going back from `Me` screen was needed, was after running `e2eLoginWithSelfHostedAccount`. It can be seen from the logs that all the UI tests were failing after this test - since to "log out" from the self-hosted account, we need to open `Me` screen, see that it's a self-hosted account, and then go back to `My sites` and delete a site.

**Comment by tiagomar** [comment]
In WordPress app, as long as `Me` is a tab, we probably want to remove the back arrow on top left corner? It will be needed when `Me` is moved to `Menu` though.

**Comment by AjeshRPai** [comment]
Hey @ravishanker 

The people icon and the me option in menu is fixed, but I found the same bug as @tiagomar reported here. the back button is visible on wordpress app in static posters phase. In an effort to merge this change before the code freeze on monday and also because your are AFK, I went ahead and fixed it with the commit [60fb503](https://github.com/wordpress-mobile/WordPress-Android/pull/18606/commits/60fb50372a42f39edd5e2b06d3fbf10bd40fe5f8)

@zwarm am adding you as a reviewer, once you have reviewed the changes i have done you can go ahead and merge it. 👍🏼 

I have tested the jetpack app and wordpress app for all other scenarios, everything looks good to me. 🟢 

@tiagomar if you can review it once again, it would be awesome 🙌🏼

**Review by AjeshRPai — APPROVED** [review]

**Review by zwarm — APPROVED** [review]
@AjeshRPai - I've verified the following
- The back button does not show in JP/WP when "me" is a tab
- The back button shows in WP when "me" is in the menu
👍 

@ravishanker @osullivanchris - love this change. 🙇‍♀️

**Comment by tiagomar** [comment]
> @tiagomar if you can review it once again, it would be awesome 🙌🏼

It was all good, @AjeshRPai! 👍 

> Me tab is reading the content description correctly. When on Me screen, it is reading the MySite description. Still investigating where that is coming from!

Yep, I've filed a ticket (#18698 ) for better tracking.

**Pull request merged**